alphadia/workflow/managers/raw_file_manager.py

```diff
@@ -0,0 +1,151 @@
+"""Manager handling the raw data file and its statistics."""
+
+import logging
+import os
+
+import numpy as np
+
+from alphadia.data import alpharaw_wrapper, bruker
+from alphadia.workflow.config import Config
+from alphadia.workflow.manager import BaseManager
+
+logger = logging.getLogger()
+
+
+class RawFileManager(BaseManager):
+    def __init__(
+        self,
+        config: None | Config = None,
+        path: None | str = None,
+        load_from_file: bool = False,
+        **kwargs,
+    ):
+        """Contains and updates timing information for the portions of the workflow."""
+        self.stats = {}  # needs to be before super().__init__ to avoid overwriting when loading
```

Comment:
Is there a solution for this? sounds like a potential pitfall?

Reply:
indeed I fell into that pit yesterday (LLMs that hallucinate that pickling a dictionary with mixed data types is problematic didn't really help ;-)) ..
option # 1

`if not hasattr(self, '_config'):
    self._config: Config = config`
ugly, uses hasattr

option # 2:
`if not load_from_file: 
    self.stats = {}`
at least this makes it more explicit than just the comment but does not prevent new users falling into the pit..

option # 3: 
each class has an explicit list of attributes to pickle, and some mechanism to instantiate those only in case its not pickled .. overkill imho..


I would go for option 2 and add a comment to the super class as well

Reply:
Sounds good 🤔 Ideally I would like to move away from pickling at some point, which could include this. Its just not an important feature.